### PR TITLE
clean up cc form, add debug values

### PR DESCRIPTION
**Pull request by jdungan**
Moves the test cc value to only fill in when in debug mode.

Also, ditched the icon rather than add motion ui just for the spinning.

### CODE REVIEW

File: static/js/home-app.js [hunk 1]
```diff
@@ -4,72 +4,56 @@ $(document).ready(function(){
...
 $(window).load(function () {
-  var codesy = {user:{}}
-
-  Stripe.setPublishableKey('pk_test_NBTdQSXW5HcsEVNgzOZmmZng');
-
-  $('#refreshing').hide();
-
-  if ($("#codesy_user_id").length > 0) {
+    var codesy = {user:{}}
+    Stripe.setPublishableKey('pk_test_NBTdQSXW5HcsEVNgzOZmmZng');
```
Comment (groovecoder) [inline static/js/home-app.js]: This should come from env var right?

Comment (jdungan) [inline static/js/home-app.js]: hmmm.  The .js is static, but we could put save the settings.key into the global scope as part of the template...?

Comment (groovecoder) [inline static/js/home-app.js]: Here's how I like to get env vars thru settings into static JS ...

`codesy/context_processors.py`:
```
from django.conf import settings


def conf_settings(request):
    return {'settings': settings}
```
`codesy/settings.py`:
```
TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [path('templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',

                'codesy.context_processors.conf_settings'
            ],
        },
    },
]
```
`base.html`:
```
<html id="codesy-html" data-stripe_key="{{ settings.STRIPE_KEY }}">
```
`static/js/home-app.js`:
```
var stripe_key = document.getElementById("codesy-html").dataset.stripe_key;
Stripe.setPublishableKey(stripe_key);
```

Comment (groovecoder) [inline static/js/home-app.js]: Of course, jQuery has the syntax sugar for the `.js` part

Comment (groovecoder) [inline static/js/home-app.js]: Note: that context processors will make **all** `settings.*` available to **all** templates, so it's handy to have around anyway.

**Comment by jdungan** [comment]
I went the route of adding it to the context and then adding this:

      <script type="text/javascript">
            Stripe.setPublishableKey('{{ stripe_context.STRIPE_PUBLIC_KEY }}');
      </script>

File: static/js/home-app.js (hunk 1 above)
Comment (jdungan) [inline static/js/home-app.js]: I like adding all the settings into the context.

**Comment by groovecoder** [comment]
Yeah, I find it easier to just add `settings` to all contexts across all templates.

File: codesy/views.py [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
+        ctx['cc_debug'] = cc_debug
+    return ctx
 
 
 class Home(TemplateView):
     template_name = 'home.html'
 
     def get_context_data(self, **kwargs):
         ctx = super(Home, self).get_context_data(**kwargs)
+        ctx = add_cc_debug(ctx)
```
Comment (groovecoder) [inline codesy/views.py]: Can we make this:
```
ctx['cc_debug'] = cc_debug()
```
So we're explicit about what key is going into the context in `get_context_data`?

File: codesy/templates/card_info.html [hunk 3]
```diff
@@ -11,6 +11,9 @@
 {% block scripts %}
    {% if request.user.is_authenticated %}
       <script type="text/javascript" src="https://js.stripe.com/v2/"></script>
-      <script src="{% static "js/home-app.js" %}" ></script>
+      <script type="text/javascript">
+            Stripe.setPublishableKey('{{ settings.STRIPE_PUBLIC_KEY }}');
```
Comment (groovecoder) [inline codesy/templates/card_info.html]: Can we move this line into `stripe-app.js` too?

**Comment by groovecoder** [comment]
Re-based and resubmitted at https://github.com/codesy/codesy/pull/343

**Pull request closed without merging**